### PR TITLE
[PW-8295] Fix giftcard capture mode for adyen_payment table

### PR DESCRIPTION
**Pull request by candemiralp**
## Summary

Method identifier is being obtained from the webhook's `paymentMethod` object instead of `orderTransaction`. That allows setting different capture modes for partial payments.

## Tested scenarios

- Giftcard payments
- Giftcard + Klarna payments

### CODE REVIEW

File: src/Service/PluginPaymentMethodsService.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        return $paymentMethods->getElements();
+    }
+
+    /**
+     * @param string $paymentMethod
+     * @return string
+     * @throws AdyenException
+     */
+    public function getHandlerIdentifierFromTxVariant(string $paymentMethod): string
+    {
```
Comment (peterojo) [inline src/Service/PluginPaymentMethodsService.php:76]: this method will not catch edge case variants like `visa_amazonpay`, we might end up going down a rabbit hole trying to map all tx_variants in order to get identify adyen payments.
What do you think @candemiralp ?

File: src/Service/PluginPaymentMethodsService.php [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            foreach ($pluginPaymentMethods as $pluginPaymentMethod) {
+                if (($pluginPaymentMethod->getHandlerIdentifier()::getPaymentMethodCode() === 'giftcard' &&
+                    $pluginPaymentMethod->getHandlerIdentifier()::getBrand() === $paymentMethod) ||
+                    $pluginPaymentMethod->getHandlerIdentifier()::getPaymentMethodCode() === $paymentMethod) {
+                    return $pluginPaymentMethod->getHandlerIdentifier();
+                }
+            }
+        }
+
+        throw new AdyenException(sprintf("Payment method %s not found!", $paymentMethod));
```
Comment (peterojo) [inline src/Service/PluginPaymentMethodsService.php:91]: this exception is not caught in `ProcessNotificationsHandler` so this scenario will break the scheduled task

File: src/Service/PluginPaymentMethodsService.php (hunk 1 above)
Comment (candemiralp) [reply src/Service/PluginPaymentMethodsService.php:76]: You are totally right, let's discuss it on a call.

File: src/Service/PluginPaymentMethodsService.php (hunk 2 above)
Comment (candemiralp) [reply src/Service/PluginPaymentMethodsService.php:91]: I tested it and It's been caught by handler's catch block and notification is being marked as done after 5th attempt.

**Pull request merged**